lnst/Recipes/ENRT/ConfigMixins/FirewallMixin.py

```diff
@@ -0,0 +1,197 @@
+from lnst.Recipes.ENRT.ConfigMixins import BaseSubConfigMixin
+from tempfile import mkstemp
+from os import close, write, unlink
+import logging
+import copy
+
+class FirewallMixin(BaseSubConfigMixin):
+    """
+    A config mixin to apply custom firewall rulesets on hosts.
+    Do not inherit directly, use one of the derived classes below instead.
+    """
+
+    @property
+    def firewall_rulesets(self):
+        """
+        This property holds a dictionary of firewall rulesets in textual
+        representation, indexed by the host it should be applied to. A typical
+        use is:
+
+        { self.matched.host1: <host1 ruleset>,
+          self.matched.host2: <host2 ruleset> }
+
+        The rulesets will be applied by a derived class's _apply_ruleset()
+        method, i.e. typically fed into 'nft -f' or 'iptables-restore'.
+        """
+        return {}
+
+    def _apply_ruleset(self, host, path):
+        raise NotImplementedError("Method _apply_ruleset must be defined by a child class.")
```

Comment:
What about marking this as [`abstractmethod`](https://docs.python.org/3/library/abc.html#abc.abstractmethod)? Imo,  it's safer as it doesn't allow you to even instantiate the class. Also, `abstractmethods` would make it easier for static analysis tools.

Reply:
DONE, thanks for pointing me at ABC and @AbstractMethod. I wasn't aware such "fancy" stuff exists in Python.